crates/red_knot_project/src/metadata/value.rs

```diff
@@ -0,0 +1,173 @@
+use ruff_db::system::{System, SystemPath, SystemPathBuf};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::cell::RefCell;
+use std::fmt;
+use std::hash::{Hash, Hasher};
+use std::sync::Arc;
+
+use crate::combine::Combine;
+use crate::Db;
+
+#[derive(Clone, Debug)]
+pub enum ValueSource {
+    /// Value loaded from a project's configuration file.
+    ///
+    /// Ideally, we'd use [`ruff_db::files::File`] but we can't because the database hasn't been
+    /// created when loading the configuration.
+    File(Arc<SystemPathBuf>),
+    /// The value comes from a CLI argument, while it's left open if specified using a short argument,
+    /// long argument (`--extra-paths`) or `--config key=value`.
+    Cli,
+}
```

Comment:
Minor: Do we need this for something else besides path anchoring? If not, I think I would prefer if this struct were a little more tailored to this use case in terms of naming. This would also make the comments a bit simpler(?). Something like
```rs
pub enum PathAnchor {
    /// …
    RelativeTo(Arc<SystemPathBuf>)
    /// …
    RelativeToCwd
}
```

Reply:
The idea is to also use it in diagnostic. E.g. the rule `x` that you specified via the CLI is unknown or this rule is checked because it was enabled via the CLI.

Reply:
Makes sense, thank you.